selftests/functional/serial/requirements.py

```diff
@@ -294,3 +302,28 @@ def test_job_dependency(self):
                     "vim-common",
                     result.stdout_text,
                 )
+
+    @skipUnless(os.getenv("CI"), skip_install_message)
+    def test_dependency_recipe(self):
+        with script.Script(
+            os.path.join(self.tmpdir.name, "test_multiple_success.py"),
+            TEST_WITHOUT_DEPENDENCY,
+        ) as test:
+            dependency_recipe_data = DEPENDENCY_RECIPE.replace(
```

Comment:
This works, but, I wonder why you didn't go with standard Python formatting strings here.

Reply:
I am not sure what do you mean, can you please give me an example? :sweat_smile:

Reply:
I mean:

```python
DEPENDENCY_RECIPE_FMT = """
{{
  "kind": "avocado-instrumented",
  "uri": "{path}",
  "kwargs": {{"dependencies": [{{"type": "package", "name": "hello"}}]}}
}}
"""
```

And then:

```python
DEPENDENCY_RECIPE_FMT.format(path=path)
```

It's just a bit more Pythonic and predictable IMO.

Reply:
Oh, right. This is definitely much better. I will send an update.